shared/src/types/token.rs

```diff
@@ -82,6 +82,13 @@ impl Amount {
             micro: change as u64,
         }
     }
+
+    /// Checked addition on amounts
+    pub fn checked_add(&self, amount: &Amount) -> Option<Self> {
+        self.micro
+            .checked_add(amount.micro)
+            .map(|micro| Self { micro })
+    }
```

Comment:
+1, I need to start using this as well, have opened https://github.com/anoma/namada/issues/748 to expand and push back to `main`